runtime/permissions/lib.rs

```diff
@@ -3331,4 +3520,114 @@ mod tests {
     )
     .is_err());
   }
+
+  #[test]
+  fn test_net_descriptor_parsing() {
+    // Parsing host address without a port
+    assert_eq!(
+      NetDescriptor::from_cli_arg("deno.land.").unwrap(),
+      NetDescriptor(Host::FQDN(FQDN::from_str("deno.land").unwrap()), None)
+    );
+    // Parsing host address with a port
+    assert_eq!(
+      NetDescriptor::from_cli_arg("deno.land:80").unwrap(),
+      NetDescriptor(Host::FQDN(FQDN::from_str("deno.land").unwrap()), Some(80))
+    );
+
+    // Parsing an IPv4 address
+    assert_eq!(
+      NetDescriptor::from_cli_arg("127.0.0.1").unwrap(),
+      NetDescriptor(Host::Ipv4(Ipv4Addr::new(127, 0, 0, 1)), None)
+    );
+    // Parsing an IPv4 address with a port
+    assert_eq!(
+      NetDescriptor::from_cli_arg("127.0.0.1:80").unwrap(),
+      NetDescriptor(Host::Ipv4(Ipv4Addr::new(127, 0, 0, 1)), Some(80))
+    );
+
+    // Parsing an IPv6 address
+    assert_eq!(
+      NetDescriptor::from_cli_arg("[2606:4700:4700::1111]").unwrap(),
+      NetDescriptor(
+        Host::Ipv6(Ipv6Addr::new(0x2606, 0x4700, 0x4700, 0, 0, 0, 0, 0x1111)),
+        None
+      )
+    );
+    // Parsing an IPv6 address with a port
+    assert_eq!(
+      NetDescriptor::from_cli_arg("[2606:4700:4700::1111]:80").unwrap(),
+      NetDescriptor(
+        Host::Ipv6(Ipv6Addr::new(0x2606, 0x4700, 0x4700, 0, 0, 0, 0, 0x1111)),
+        Some(80)
+      )
+    );
+
+    // Parsing a URL with a host
+    assert_eq!(
+      NetDescriptor::from_cli_arg("https://github.com/denoland/").unwrap(),
```

Comment:
Can you revert this new functionality? We should try to make this PR as similar to what it was like before minus the panic.

Reply:
OK